lib/floki/finder.ex

```diff
@@ -95,7 +95,12 @@ defmodule Floki.Finder do
           nil -> [n|res_acc]
           _ ->
             {_, _, children_nodes} = n
-            traverse(children_nodes, sibling_nodes, selector.combinator.selector, res_acc)
+
+            if selector.combinator.match_type == :child do
+              traverse_child(children_nodes, sibling_nodes, selector.combinator.selector, res_acc)
+            else
+              traverse(nodes, sibling_nodes, selector, res_acc)
```

Comment:
@philss that didn't help. I think I'm doing it wrong when passing `nodes`, as `nodes` is something Enum.reduce is still reducing on